pkg/bgpv1/agent/controller.go

```diff
@@ -188,12 +174,22 @@ func (c *Controller) Run(ctx context.Context) {
 		})
 	)
 
-	// add an initial signal to kick things off
-	c.Sig.Event(struct{}{})
-
 	l.Info("Cilium BGP Control Plane Controller now running...")
 	for {
 		select {
+		case ev, ok := <-c.CiliumNodeResource.Events(ctx):
```

Comment:
Wouldn't the current approach reinitialize Events() on each loop? I'd suggest rewriting this like below :

```
ciliumNodeCh := c.CiliumNodeResource.Events(ctx)

for {
  select {
    case ev, ok := <- ciliumNodeCh:
   ...
..
```

Reply:
Updated.